src/pyspiffe/spiffe_id/trust_domain.py

```diff
@@ -1,11 +1,59 @@
+from urllib.parse import urlparse
+
+from pyspiffe import SPIFFE_SCHEME
+
+EMPTY_DOMAIN_ERROR = 'Trust domain cannot be empty'
+SCHEME_SUFFIX = '://'
+
+
 class TrustDomain(object):
     """
     Represents the name of a SPIFFE trust domain (e.g. 'domain.test').
 
-    :param trust_domain: The name of the Trust Domain
+    Args:
+        name(str): The name of the Trust Domain
+
+    Raises:
+        ValueError: is the name of the trust domain is empty, has a port, or contains an invalid scheme.
+
     """
 
-    name: str
+    def __init__(self, name: str):
+        self.__set_name(name)
+
+    def __str__(self):
+        return self.__name
+
+    def __eq__(self, other):
+        if isinstance(other, TrustDomain):
+            return self.__name == other.get_name()
+        return False
+
+    def __set_name(self, name):
+        if name == '' or name is None:
+            raise ValueError(EMPTY_DOMAIN_ERROR)
+
+        name = self.__normalize(name)
+        uri = urlparse(name)
+        self.__validate_uri(uri)
+        self.__name = uri.hostname.lower().strip()
+
+    def get_name(self):
+        return self.__name
+
+    @staticmethod
+    def __normalize(name: str):
+        if not SCHEME_SUFFIX in name:
+            name = SPIFFE_SCHEME + SCHEME_SUFFIX + name
+        return name
 
-    def __init__(self, trust_domain: str):
-        self.name = trust_domain
+    @staticmethod
+    def __validate_uri(uri):
+        if not uri.scheme == SPIFFE_SCHEME:
+            raise ValueError(
+                'Trust domain: invalid scheme: expected {}'.format(SPIFFE_SCHEME)
```

Comment:
I'll start to be super nit picky so from now on you can ignore me 😝 
what if we try to be consistent and extract this error to a constant, same as you are already doing with `EMPTY_DOMAIN_ERROR`, and the same with `Trust domain: port is not allowed` a few lines below

Reply:
I didn't extract it because my logic was "extract constant when it is used in multiple places"